src/Common/src/Interop/OSX/Interop.CoreFoundation.cs

```diff
@@ -0,0 +1,111 @@
+        /// </summary>
+        /// <param name="allocator">Should be IntPtr.Zero</param>
+        /// <param name="str">The string to get a CFStringRef for</param>
+        /// <param name="encoding">The encoding of the str variable. This should be UTF 8 for OS X</param>
+        /// <returns>Returns a pointer to a CFString on success; otherwise, returns IntPtr.Zero</returns>
+        [DllImport(CoreFoundationLibrary)]
+        internal static extern CFStringRef CFStringCreateWithCString(
+            IntPtr allocator, 
+            string str, 
+            CFStringBuiltInEncodings encoding);
+
+        /// <summary>
+        /// Creates a pointer to an unmanaged CFArray containing the input values. Folows the "Create Rule" where if you create it, you delete it.
+        /// </summary>
+        /// <param name="allocator">Should be IntPtr.Zero</param>
+        /// <param name="values">The values to put in the array</param>
+        /// <param name="numValues">The number of values in the array</param>
+        /// <param name="callbacks">Should be IntPtr.Zero</param>
+        /// <returns>Returns a pointer to a CFArray on success; otherwise, returns IntPtr.Zero</returns>
+        [DllImport(CoreFoundationLibrary)]
+        internal static extern CFArrayRef CFArrayCreate(
+            IntPtr allocator,
+            [MarshalAs(UnmanagedType.LPArray)]
+            IntPtr[] values,
+            ulong numValues,
+            IntPtr callbacks);
+
+        /// <summary>
+        /// Retrieves the RunLoop associated with the current thread; all threads automatically have a RunLoop.
+        /// Follows the "Get Rule" where you do not own the object unless you CFRetain it; in which case, you must also CFRelease it as well.
+        /// </summary>
+        /// <returns>Returns a pointer to a CFRunLoop on success; otherwise, returns IntPtr.Zero</returns>
+        [DllImport(CoreFoundationLibrary)]
+        internal static extern CFRunLoopRef CFRunLoopGetCurrent();
+
+        /// <summary>
+        /// Decrements the reference count on the specified object and, if the ref count hits 0, cleans up the object.
+        /// </summary>
+        /// <param name="ptr">The pointer on which to decrement the reference count.</param>
+        [DllImport(CoreFoundationLibrary)]
+        internal extern static void CFRelease(IntPtr ptr);
+
+        /// <summary>
+        /// Starts the current thread's RunLoop. If the RunLoop is already running, creates a new, nested, RunLoop in the same stack.
+        /// </summary>
+        [DllImport(CoreFoundationLibrary)]
+        internal extern static void CFRunLoopRun();
+
+        /// <summary>
+        /// Notifies a RunLoop to stop and return control to the execution context that called CFRunLoopRun
+        /// </summary>
+        /// <param name="rl">The RunLoop to notify to stop</param>
+        [DllImport(CoreFoundationLibrary)]
+        internal extern static void CFRunLoopStop(CFRunLoopRef rl);
+    }
+}
```

Comment:
Files should end with a newline... that's what the little symbol at the end of the line is indicating.